core/shared/platform/windows/win_socket.c

```diff
@@ -45,6 +58,16 @@ os_socket_create(bh_socket_t *sock, bool is_ipv4, bool is_tcp)
         return BHT_ERROR;
     }
 
+    *(sock) = wasm_runtime_malloc(sizeof(windows_handle));
```

Comment:
What's the underlying type of the `bh_socket_t` for windows? Is it a pointer to pointer or `intptr_t`? Otherwise we might loose part of the pointer.